Armadillo/src/main/java/com/scribd/armadillo/encryption/SecureStorage.kt

```diff
@@ -59,27 +71,40 @@ internal class ArmadilloSecureStorage @Inject constructor(
     }
 
     override fun saveDrmDownload(context: Context, audioUrl: String, drmDownload: DrmDownload) {
-        val key = getDrmDownloadKey(audioUrl, drmDownload.drmType)
+        val alias = getDrmDownloadAlias(audioUrl, drmDownload.drmType)
         val value = Base64.encodeToString(Json.encodeToString(drmDownload).toByteArray(StandardCharsets.UTF_8), Base64.NO_WRAP)
-        drmDownloadStorage.edit().putString(key, value).apply()
+        secureDrmStorage.edit().putString(alias, value).apply()
     }
 
-    override fun getDrmDownload(context: Context, audioUrl: String, drmType: DrmType): DrmDownload? =
-        drmDownloadStorage.getString(getDrmDownloadKey(audioUrl, drmType), null)?.decodeToDrmDownload()
+    override fun getDrmDownload(context: Context, audioUrl: String, drmType: DrmType): DrmDownload? {
+        val alias = getDrmDownloadAlias(audioUrl, drmType)
+        var download = secureDrmStorage.getString(alias, null)?.decodeToDrmDownload()
+        if (download == null && legacyDrmStorage.contains(alias)) {
+            //migrate old storage to secure storage
+            val downloadValue = legacyDrmStorage.getString(alias, null)
+            download = downloadValue?.decodeToDrmDownload()
+            secureDrmStorage.edit().putString(alias, downloadValue).apply()
+            legacyDrmStorage.edit().remove(alias).apply()
+        }
+        return download
+    }
 
     override fun getAllDrmDownloads(context: Context): Map<String, DrmDownload> =
-        drmDownloadStorage.all.keys.mapNotNull { key ->
-            drmDownloadStorage.getString(key, null)?.let { drmResult ->
+        legacyDrmStorage.all.keys.mapNotNull { key ->
+            legacyDrmStorage.getString(key, null)?.let { drmResult ->
                 key to drmResult.decodeToDrmDownload()
             }
         }.toMap()
```

Comment:
With the switch to secure storage, this method will return an empty map. We need to use the secure drm storage

Reply:
Oh, good catch. Do you need to actually check _both_?